open-learning-exchange/planet
# Build without QEMU Binary (fixes #651 #621)

fixes #651 #621 
Should work As expected.
Please test it on Raspberry Pi. Thanks

How to test:
1. You will have to create a `planet-test.yml` file with the following contents:

```
services:
  couchdb:
    expose:
      - 5984
    image: treehouses/rpi-couchdb:2.1.1
    ports:
      - "2200:5984"
      - "2201:5986"
  db-init:
    image: treehouses/planet:rpi-db-init-0.1.2-build-without-qemu-binary-e934374f
    depends_on:
      - couchdb
    environment:
      - COUCHDB_HOST=http://couchdb:5984
  planet:
    image: treehouses/planet:rpi-0.1.2-build-without-qemu-binary-e934374f
    ports:
      - "80:80"
    environment:
      - MULTIPLE_IPS=true
      - HOST_PROTOCOL=http
      - DB_HOST=127.0.0.1
      - DB_PORT=2200
      - CENTER_ADDRESS=earth.np.ole.org:5986
    depends_on:
      - couchdb
version: "2"
```

2. Then use the following command to spawn your environment for the first time.


```
docker-compose -f planet-test.yml -p planet-test up -d --build
```

See if the docker containers running

```
docker ps
```

You'll see you containers like this

```
CONTAINER ID        IMAGE               COMMAND                  CREATED              STATUS              PORTS                                                                NAMES
ea3b914c3193        planetdev_planet    "/bin/sh -c 'bash ..."   About a minute ago   Up 58 seconds       0.0.0.0:3000->3000/tcp                                               planetdev_planet_1
57f30698ccda        klaemo/couchdb      "tini -- /docker-e..."   About a minute ago   Up About a minute   4369/tcp, 9100/tcp, 0.0.0.0:2200->5984/tcp, 0.0.0.0:2201->5986/tcp   planetdev_couchdb_1
```

Connect to your `planetdev_planet` with

```
docker logs {{id}}
```

in this case

```
docker logs ea3b914c3193 -f
```

press `CTRL+C` to exit logs view

3. When you're done, you can do the following command

```
docker-compose -f planet-test.yml -p planet-test stop
```

4. When you go back to code

```
docker-compose -f planet-test.yml -p planet-test start
```

5. When you have to delete the environment

```
docker-compose -f planet-test.yml -p planet-test down
```
